CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+
+cmake_minimum_required(VERSION 3.20)
+set(PROJECT rootbeer)
+project(${PROJECT})
+
+include(cmake/luajit.cmake)
+
+set(SOURCES
+    "src/main.c"
+    "src/config/load.c"
+    "src/config/module.c"
+    "src/config/functions/debug_test.c"
+
+    "src/cli/parse.c"
+    "src/cli/commands.c"
+    "src/cli/commands/compose.c"
+    "src/cli/commands/store.c"
+
+    "src/store/create.c"
+    "src/store/current.c"
+    "src/store/manage.c"
+    "src/store/read.c"
+    "src/store/view.c"
+)
+
+add_executable(${PROJECT}
+    ${SOURCES}
+)
+
+target_include_directories(${PROJECT} PUBLIC
+    "include/"
+    "${LUAJIT_INCLUDE_DIRS}"
+)
+
+# required for linking with built externals
+target_link_directories(${PROJECT} PUBLIC
+     "${CMAKE_BINARY_DIR}"
+     "${CMAKE_BINARY_DIR}/lib"
+)
+
+target_link_libraries(${PROJECT} PUBLIC
+    "${LUAJIT_LIBRARIES}"
+)
+
+add_dependencies(${PROJECT} luajit-external)
```

Comment:
just ensures that it actually downloads the external first.

Probably not the best long term solution. Should abstract to some function or macro in a general include that takes in a target (${PROJECT}) in this case which would automatically add the dependency and link its libraries to the target so we don't have this "random" stuff like target_link_directories

Reply:
kinda like the find2 we found in neovims cmake